x-pack/plugins/siem/public/components/load_more_table/index.tsx

```diff
@@ -79,21 +79,14 @@ export class LoadMoreTable<T> extends React.PureComponent<BasicTableProps<T>, Ba
     paginationLoading: false,
   };
 
-  public componentDidUpdate(prevProps: BasicTableProps<T>) {
-    const { paginationLoading, isEmptyTable } = this.state;
-    const { loading, pageOfItems } = this.props;
-    if (paginationLoading && prevProps.loading && !loading) {
-      this.setState({
-        ...this.state,
-        paginationLoading: false,
-      });
-    }
-    if (isEmpty(prevProps.pageOfItems) && !isEmpty(pageOfItems) && isEmptyTable) {
-      this.setState({
-        ...this.state,
+  static getDerivedStateFromProps(props: BasicTableProps<any>, state: BasicTableState) {
```

Comment:
Can this `any` be templated as the previous `BasicTableProps<T>` was?

Reply:
I am getting a weird error when I do it